### ISSUE TITLE
Automatic Backport PR Tags

### ISSUE THREAD

**Issue by nfuden** [solo-io/gloo#6734]
### Is your feature request related to a problem? Please describe.

Backports are painful and take alot of time especially with multiple repos.

### Describe the solution you'd like

In general it would be nice to have as much of the backport process automated. 
For this first chunk of effort we propose starting with a PR tag that will auto cherry pick and open Backport PRs based on some tag.
This is similar to what is in place for isitio.

**Comment by sam-heilbron** [comment]
We have had some previous attempts at introducing this:
- [Nathan] https://github.com/solo-io/solo-projects/pull/5576 
- [Nina] https://github.com/solo-io/gloo/pull/9641

I think for the most part we agree that some form of automation would really help. We could explore this in phases:
- Local tool that devs can run from their machine
- Manual workflow that devs can run in CI
- Some form of automation to run the workflow after a PR merges based on some criteria

I'm not sure what we want to bite off first. One consideration is just that having some toil to our backport process _can_ at times be a good thing. While backports are good to perform when we are introducing fixes, they also include a certain amount of risk that should be evaluated each time.

cc @timflannagan for [recent slack question](https://solo-io-corp.slack.com/archives/G01EERAK3KJ/p1724270803962779) about the state of this work

**Comment by timflannagan** [comment]
> While backports are good to perform when we are introducing fixes, they also include a certain amount of risk that should be evaluated each time.

@sam-heilbron Good point, but I think this is better tackled with a documented process that covers the following:

- Document our backport policy. How far back do we backport CVEs/bug fixes? What about low or medium severity bugs? etc.
- Uses labels to indicate how far back GH issues need to be backported. Ex: `Needs Backport 1.17.x`. This can be added during refinement so there's less ambiguity / more signal to developers once engineering picks up an issue

I sketched that out inline, so that list doesn't have to be definitive. Having this in place helps solve "we don't want everything to be backported due to N reasons (e.g. risk, regressions patch versions, etc.) so we intentionally introduce friction in the process to help mitigate that". WYDT?
